<?xml version="1.0"?>
<!-- -->
<launch>
  <arg name = "record" default = "false"/>
  <param name = "includeUnav" value="true" type="bool"/>
  <param name = "IncludeScans" value="111111" type="string"/>
  <param name = "ref_sensor" value="lms151_E" type="string"/>
  <!--Launch the visualization-->
  <node pkg="rviz" type="rviz" name="rviz" args="--display-config $(find free_space_detection)/config/rviz.rviz" required="true"/>
  <!-- Launch the node   -->
  <!-- <node pkg="free_space_detection" type="free_space_detection_node" name="free_space_detection" required="true" output="screen"/> -->
  <node pkg="free_space_detection" type="device_frame_publisher_node" name="device_frame_publisher" required="true" output="screen"/>
  <!-- Launch the tf publisher
  <node pkg="tf" type="static_transform_publisher" name="root_broadcaster" args="0 0 0 0 0 0 1 map root 100" /><node pkg="tf" type="static_transform_publisher" name="lms151_D_broadcaster" args="0 -0.8 0 0 0 -0.43 1 root lms151_D 100" /><node pkg="tf" type="static_transform_publisher" name="lms151_E_broadcaster" args="0 0.8 0 0 0 0.43 1 root lms151_E 100" /><node pkg="tf" type="static_transform_publisher" name="ldmrs0_broadcaster" args="0 0 0 0 0 0 1 root ldmrs0 100" /><node pkg="tf" type="static_transform_publisher" name="ldmrs1_broadcaster" args="0 0 0 0 0 0 1 root ldmrs1 100" /><node pkg="tf" type="static_transform_publisher" name="ldmrs2_broadcaster" args="0 0 0 0 0 0 1 root ldmrs2 100" /><node pkg="tf" type="static_transform_publisher" name="ldmrs3_broadcaster" args="0 0 0 0 0 0 1 root ldmrs3 100" /> -->
  <!--Launch the recorder-->
  <node name="rosbag_record" pkg="rosbag" type="record" output="screen" if="$(arg record)" args=" /ld_rms/scan0 /ld_rms/scan1  /ld_rms/scan2 /ld_rms/scan3 /lms151_D_scan /lms151_E_scan /camera/image_color -o $(find free_space_detection)/laserData/"/>
  <!--Launch the robot model-->
  <include file="$(find free_space_detection)/launch/robotmodel.launch" output="log"/>
</launch>
